File: 08-download-a-file-revisited/java/src/test/java/DownloadFileRevisited.java
package src.test.java;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import static org.hamcrest.CoreMatchers.*;
import static org.hamcrest.MatcherAssert.assertThat;
import org.apache.http.HttpResponse;
import org.apache.http.client.HttpClient;
import org.apache.http.client.methods.HttpHead;
import org.apache.http.impl.client.HttpClientBuilder;

public class DownloadFileRevisited {
    WebDriver driver;

    @Before
    public void setUp() throws Exception {
        driver = new FirefoxDriver();
    }

    @After
    public void tearDown() throws Exception {
        driver.quit();
    }

    @Test
    public void downloadFileRevisitedTest() throws Exception {
        driver.get("http://the-internet.herokuapp.com/download");
        String link = driver.findElement(By.cssSelector(".example a:nth-of-type(1)")).
                                            getAttribute("href");

        HttpClient httpClient = HttpClientBuilder.create().build();
        HttpHead request = new HttpHead(link);
        HttpResponse response = httpClient.execute(request);
        String contentType = response.getFirstHeader("Content-Type").getValue();
        int contentLength = Integer.parseInt(response.getFirstHeader("Content-Length").getValue());

        assertThat(contentType, is("application/octet-stream"));
        assertThat(contentLength, is(not(0)));
    }

}

File: 08-download-a-file-revisited/kotlin/src/test/kotlin/DownloadFileRevisited.kt
import org.openqa.selenium.By
import org.openqa.selenium.firefox.FirefoxDriver
import kotlin.test.assertEquals
import kotlin.test.AfterTest
import kotlin.test.Test
import kotlin.test.assertNotEquals
import org.apache.http.client.methods.HttpHead
import org.apache.http.impl.client.HttpClientBuilder

class DownloadFileRevisited {
    private val testUrl = "http://the-internet.herokuapp.com/download"

    private val driver by lazy {
        FirefoxDriver()
    }

    @AfterTest
    fun tearDown() {
        // Quit session (closes browser)
        driver.quit()
    }

    @Test
    @Throws(Exception::class)
    fun downloadFileRevisitedTest() {
        driver.get(testUrl)
        val link = driver.findElement(By.cssSelector(".example a:nth-of-type(1)")).getAttribute("href")
        val httpClient = HttpClientBuilder.create().build()
        val request = HttpHead(link)
        val response = httpClient.execute(request)
        val contentType = response.getFirstHeader("Content-Type").getValue()
        val contentLength = Integer.parseInt(response.getFirstHeader("Content-Length").getValue())

        assertEquals(
            "application/octet-stream",
            contentType
        )

        assertNotEquals(
            0,
            contentLength
        )
    }
}